File: dart/test/sort/sort_test.dart
import 'package:kata/sort/sort.dart';
import 'package:test/test.dart';

void main() {
  test('is sorted', () {
    expect([3, 2, 1].isSorted, false);
    expect([1, 2, 3].isSorted, true);
  });
  test('quick sort', () {
    expect([3, 1, 2, 5, 4, 10].quickSort().isSorted, true);

    expect([5, 4, 3, 4].quickSort(), [3, 4, 4, 5]);
  });

  test('is subset of', () {
    final a = [5, 2, 3, 1, 4];
    final b = [2, 1];
    expect(b.isSubsetOf(a), true);
    expect(a.isSubsetOf(b), false);

    final c = [1, 2, 3];
    final d = [4, 5, 6];

    expect(c.isSubsetOf(d), false);
    expect(d.isSubsetOf(c), false);
  });
}

File: kotlin/test/sort/SortTest.kt
package sort

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

internal class SortTest {
    @Test
    fun sortTest() {
        val testList = listOf(2, 1, 3)

        assertEquals(false, testList.isSorted())
        assertEquals(listOf(1, 2, 3), testList.quickSort())
    }

    @Test
    fun sortTest2() {
        val testList = listOf(3, 123941823, 13, 12394, 12381, 1, 48, 0)

        assertEquals(false, testList.isSorted())
        assertEquals(listOf(0, 1,3, 13, 48, 12381, 12394, 123941823), testList.quickSort())
    }

    @Test
    fun sortTest3() {
        val testList = listOf(5, 1, 4, 2, 5)

        assertEquals(listOf(1, 2, 4, 5, 5), testList.quickSort())
    }

    @Test
    fun subsetTest() {
        val a = listOf(5, 2, 4, 1)
        val b = listOf(1, 2)

        assertEquals(true, b.isSubsetOf(a))
    }

    @Test
    fun subsetTest2() {
        val a = listOf(1, 2, 3)
        val b = listOf(4, 5)

        assertEquals(false ,b.isSubsetOf(a))
    }
}